### PR TITLE
r/rds_cluster: enable data API

### PR DESCRIPTION
**Pull request by svanharmelen**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Add an option to enable the [data API](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/data-api.html) for serverless deployments.

Next to running some manual test, I also executed the new acceptance test:

image

### CODE REVIEW

File: aws/resource_aws_rds_cluster_test.go [hunk 1]
```diff
@@ -2139,6 +2163,28 @@ resource "aws_rds_cluster" "default" {
...
 
+func testAccAWSClusterConfigWith_EnableDataAPI(n int) string {
+	return fmt.Sprintf(`
+resource "aws_rds_cluster" "test" {
+  cluster_identifier  = "tf-aurora-cluster-%d"
+  availability_zones  = ["us-west-2a", "us-west-2b", "us-west-2c"]
+  database_name       = "mydb"
+  master_username     = "foo"
+  master_password     = "mustbeeightcharaters"
+	engine_mode         = "serverless"
```
Comment (kislyuk) [inline aws/resource_aws_rds_cluster_test.go]: Tabs/spaces mismatch here.

Comment (svanharmelen) [reply aws/resource_aws_rds_cluster_test.go]: Fixed 👍

**Comment by svanharmelen** [comment]
@bflad could you maybe have alook at this one? Thx!

**Comment by kislyuk** [comment]
Can someone please take a look at this and merge it? It's blocking us.

**Comment by Thul95** [comment]
Would like to see this get merged!

**Comment by bflad** [comment]
Hi @svanharmelen 👋 Thank you for submitting this. It looks like this functionality was released in [version 2.45.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2450-january-17-2020) awhile ago. I apologize that this was possibly overlooked back then, but if there is anything else missing please do reach out.

**Comment by svanharmelen** [comment]
All good! Thanks 👍🏻

**Pull request closed without merging**